crates/bevy_ecs/src/reflect/component.rs

```diff
@@ -84,26 +84,27 @@ pub struct ReflectComponent(ReflectComponentFns);
 #[derive(Clone)]
 pub struct ReflectComponentFns {
     /// Function pointer implementing [`ReflectComponent::from_world()`].
-    pub from_world: fn(&mut World) -> Box<dyn Reflect>,
+    pub from_world: fn(&mut World) -> Box<dyn PartialReflect>,
     /// Function pointer implementing [`ReflectComponent::insert()`].
-    pub insert: fn(&mut EntityMut, &dyn Reflect),
+    pub insert: fn(&mut EntityMut, &dyn PartialReflect),
     /// Function pointer implementing [`ReflectComponent::apply()`].
-    pub apply: fn(&mut EntityMut, &dyn Reflect),
+    pub apply: fn(&mut EntityMut, &dyn PartialReflect),
     /// Function pointer implementing [`ReflectComponent::apply_or_insert()`].
-    pub apply_or_insert: fn(&mut EntityMut, &dyn Reflect),
+    pub apply_or_insert: fn(&mut EntityMut, &dyn PartialReflect),
     /// Function pointer implementing [`ReflectComponent::remove()`].
     pub remove: fn(&mut EntityMut),
     /// Function pointer implementing [`ReflectComponent::contains()`].
     pub contains: fn(EntityRef) -> bool,
     /// Function pointer implementing [`ReflectComponent::reflect()`].
-    pub reflect: fn(EntityRef) -> Option<&dyn Reflect>,
+    pub reflect: fn(EntityRef) -> Option<&dyn PartialReflect>,
```

Comment:
i totally agree that this can be changed without much resistance but i'm not sure we want to. currently all `ComponentId`s map 1-to-1 with Rust types, but if we (and we do want to) support dynamic component types this relationship will change, and i think there's good reasons to want to reflect those components with this api.

Reply:
I still think we need a place where we collect all the "either `PartialReflect` or `Reflect`" decisions. If you are willing to write a comment in the RFC discussions, I'll take care to update the RFC with such a new section.